### PR TITLE
Add external libs for xpu

### PR DESCRIPTION
**Pull request by Stonepia**
The user now could call the tl.math function with given libdevice_path.

The user could set the `TRITON_LIBDEVICE_PATH` to third_party libs. 

For example, `export TRITON_LIBDEVICE_PATH=...triton/python/triton/third_party/xpu/`.

### CODE REVIEW

File: python/csrc/intel_xpu_backend_for_triton.cc [hunk 1]
```diff
@@ -72,6 +72,12 @@ void init_triton_translation(py::module &m) {
           py::bytes bytes(spvbin);
           return std::move(bytes);
         });
+  m.def("add_external_libs",
+        [](mlir::ModuleOp &op, const std::vector<std::string> &names,
+           const std::vector<std::string> &paths) {
+            std::cout << "Add external libs" << std::endl;
```
Comment (chengjunlu) [inline python/csrc/intel_xpu_backend_for_triton.cc:78]: Remove std::cout

File: lib/Target/SPIRV/SPIRVTranslation.cpp [hunk 2]
```diff
@@ -232,7 +232,9 @@ static std::map<std::string, std::string> getExternLibs(spirv::ModuleOp module)
   if (!funcs.empty()) {
     std::vector<std::string> lib_names = {"libsycl-fallback-imf.spv",
                                           "libsycl-fallback-imf-fp64.spv",
-                                          "libsycl-fallback-imf-bf16.spv"};
+                                          "libsycl-fallback-imf-bf16.spv",
+                                          "libsycl-imf-dl.spv",
```
Comment (chengjunlu) [inline lib/Target/SPIRV/SPIRVTranslation.cpp:236]: What is this two lib for?

Comment (Stonepia) [reply lib/Target/SPIRV/SPIRVTranslation.cpp:236]: These two include the imf function of the following:

__devicelib_imf_abs, __devicelib_imf_pownf, __devicelib_imf_pown

File: python/csrc/intel_xpu_backend_for_triton.cc (hunk 1 above)
Comment (Stonepia) [reply python/csrc/intel_xpu_backend_for_triton.cc]: Thanks! Removed.

File: lib/Target/SPIRV/SPIRVTranslation.cpp (hunk 2 above)
Comment (EikanWang) [reply lib/Target/SPIRV/SPIRVTranslation.cpp:236]: Will it impact the linker performance? I mean we add more spv binaries, the spriv linker needs more time for optimization.

**Review by EikanWang — APPROVED** [review]

**Pull request merged**